### PR TITLE
perf(isochrones): improve isochrones performance

### LINKED ISSUE #1573: Getting Error: Gateway Timeout (HTTP 504) in RStudio
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Where did you encounter this issue?

live API

### Request URL

https://openrouteservice.org/dev/#/api-docs/v2/isochrones

### POST Request Body

```JSON
library(openrouteservice)
library(mapview)
library(tidyverse)
library(osmdata)
library(sf)
library(paletteer)
library(ggfx)

ors_api_key("5b3ce3597851110001cf62488c30fb776efd4dbb9d3a08e42f286d7d")
coordinates <- data.frame(lon = c(-0.171201), lat = c(51.464290))

cj_iso <- ors_isochrones(locations = coordinates, profile = "foot-walking", range = 6000, interval = 600, output = "sf")
```

### Response

Error: Gateway Timeout (HTTP 504).

### Current behavior

I'm using RStudio and the ORS's API for mapping isochrones between a UK station and walking time. It seems alright until I try to give the isochrones a range and an interval, in this case 6000 and 600 in that order.
 
> cj_iso <- ors_isochrones(locations = coordinates, profile = "foot-walking", range = 6000, interval = 600, output = "sf")  

It takes several minutes and the result is this message from the terminal:

> Error: Gateway Timeout (HTTP 504).

So I don't know what else to do. 

It's important to say that I contacted support and changing the interval to 600 worked out just fine, but I really needed to try 600.
Sin título

### Expected behavior

The expected is running the code just as it is, with visually the double of isochrones in map. 

### Openrouteservice Version

6.6.1

### Build date

_No response_

### Graph date

_No response_

### Forum Topic Link

_No response_

### PR DESCRIPTION
**Pull request by rtroilo**
Replacing `ConcaveHullOpenSphere` with jts own `ConcaveHull` algorithm

### Pull Request Checklist

- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes #1573.

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

**Comment by aoles** [comment]
Some preliminary benchmarks are available [here](https://rpubs.com/aoles/isochrone_performance_fix). Compared to the current master branch it looks quite promising. Apart from faster query times, the produced geometries seem to be more accurate as well. In particular, there seem to be less artifacts around long highway stretches.

**Comment by jhaeu** [comment]
Hi @aoles - the test combinationsAsGeojsonFiles was only a means to write isochrones results to files to make them visually comparable after code changes. But I think, the test could/should now be removed or at least `@Disabled `, it does not assert anything and it should not write files in the ci environment...?

**Comment by aoles** [comment]
Thanks @jhaeu! Sure, I have already addressed your concerns in https://github.com/GIScience/openrouteservice/pull/1607/commits/bce8ff555a2d6cbd622333ffbc53f339c71a51ad 👍

**Comment by MichaelsJP** [comment]
@rtroilo @aoles  @jhaeu Is this ready for review?

**Comment by aoles** [comment]
A [comprehensive benchmark](https://rpubs.com/aoles/isochrones_performance_improvement) of 60 min car isochrones split into 3, 6, and 9 intervals across 100 locations in the DACH region suggests an average performance gain of around 25%.

Additionally, there seem to be an overall improvement to the geometries. Consider the following examples, where 
- **red** is the new geometry produced by this feature branch
- **blue** is the current reference computed with the master branch.

image
<sup>Location 2</sup>

image
<sup>Location 3</sup>

image
<sup>Location 5</sup>

image
<sup>Location 7</sup>

image
<sup>Location 9</sup>

File: ors-api/src/test/java/org/heigit/ors/apitests/isochrones/ResultTest.java [hunk 1]
```diff
@@ -478,4 +489,90 @@ void testCompleteMetadata() {
                 .body("metadata.containsKey('system_message')", is(true))
                 .statusCode(200);
     }
+
+    @Disabled
```
Comment (sfendrich) [inline ors-api/src/test/java/org/heigit/ors/apitests/isochrones/ResultTest.java]: Why do we add a disabled test? Either remove the test or add a comment explaining what it is for.

File: ors-engine/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java [hunk 2]
```diff
@@ -285,40 +280,13 @@ private void markDeadEndEdges(AccessibilityMap edgeMap) {
...
-                tree.insert(env, p);
-                points.add(p);
-                treeSet.add(p);
-
-                return true;
-            }
-        }
-
-        return false;
+    public void addPoint(List<Coordinate> points, double lon, double lat) {
```
Comment (sfendrich) [inline ors-engine/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java:283]: Why can this method be simplified so much?

**Review by sfendrich — CHANGES_REQUESTED** [review]
Overal it looks good to me. I just have a few minor comments & questions.

File: ors-api/src/test/java/org/heigit/ors/apitests/isochrones/ResultTest.java (hunk 1 above)
Comment (aoles) [reply ors-api/src/test/java/org/heigit/ors/apitests/isochrones/ResultTest.java]: Good catch! It was primarily meant for debugging purposes, I've removed it.

File: ors-engine/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java (hunk 2 above)
Comment (aoles) [reply ors-engine/src/main/java/org/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java:283]: The original method of adding points to the isochrone was performing a neighbor check beforehand in order to reduce the size of the point cloud. If there were points present within the given neighborhood the point was skipped. As it turns out, however, this check implemented on top of a quad tree is a computationally expensive operation which outweights the performance gain of constructing the isochrone geometry from a reduced amount of points.

**Review by sfendrich — APPROVED** [review]
LGTM!

**Pull request merged**